src/channel.spec.ts

```diff
@@ -164,4 +165,41 @@ describe('Test the channel bridge from iFrame to admin', () => {
     expect(localeMethodMock).toHaveBeenCalledTimes(2);
     expect(fallbackLocaleMethodMock).toHaveBeenCalledTimes(2);
   });
+
+  it('should reject send with missing privileges', () => {
+    send('_privileges', {}).catch(e => {
+      const expectedError = new MissingPrivilegesError('_privileges', ['create:user', 'read:user', 'update:user', 'delete:user']);
+
+      expect(e.message).toEqual(expectedError.message)
+    })
+  });
+
+  it('should not call handle callback with missing extensions', () => {
+    const callback = jest.fn();
+    handle('_privileges', callback)
+
+    // Simulate a postMessage call from an iFrame
+    window.dispatchEvent(new Event('message'))
+
+    expect(callback).toHaveBeenCalledTimes(0)
+  });
+
+  it('should not call handle callback with missing privileges', () => {
+    const callback = jest.fn();
+    const url = 'http://example.com';
+
+    handle('_privileges', callback, {foo: {baseUrl: url, permissions: {create: ['notification']}}})
+
+    const event = new Event('message');
+
+    // Simulate a postMessage call from an iFrame with a none registered origin
+    window.dispatchEvent(event)
+
+    // Simulate a postMessage call from an iFrame
+    // @ts-expect-error
+    event.origin = url;
+    window.dispatchEvent(event)
+
+    expect(callback).toHaveBeenCalledTimes(0)
+  });
```

Comment:
Can you also add a E2E test for a positive and negative case just to make sure that it works also with iFrames. The jest tests are only checking the behavior without iFrames